app/MMB/src/components/ETicket.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+
+import * as React from 'react';
+import { graphql, PrivateApiRenderer } from '@kiwicom/mobile-relay';
+import { GeneralError } from '@kiwicom/mobile-shared';
+import { Translation } from '@kiwicom/mobile-localization';
+import PdfViewer from '@kiwicom/mobile-pdf';
+import idx from 'idx';
+
+import BookingDetailContext from '../context/BookingDetailContext';
+import type { ETicketQueryResponse } from './__generated__/ETicketQuery.graphql';
+
+export default class ETicket extends React.Component<{||}> {
+  renderInnerComponent = (renderProps: ETicketQueryResponse) => {
+    const uri = idx(renderProps, _ => _.node.assets.ticketUrl);
+
+    if (!uri) {
+      return (
+        <GeneralError
+          errorMessage={<Translation id="mmb.tickets.not_available" />}
+        />
+      );
+    }
+
+    return <PdfViewer uri={uri} />;
```

Comment:
We already have a quite similar component called `ETicketPdf.js`

This takes the uri, and stores the pdf, so you don't need to download it again each time you open it. Maybe you could re-use that component?

Reply:
I am using it by navigating to the same route you use in your other component. Hope that's how it's meant to be done.

Reply:
That's looks good to me 😄